### PR TITLE
Refactor targeting/bundle controller

### PR DESCRIPTION
**Pull request by manno**
Refers to https://github.com/rancher/fleet/issues/2110

The diff for [Refactor target package](https://github.com/rancher/fleet/pull/2111/commits/760471ba223218d5f3eb5678f1a21eb5c79ae11a) looks better with `?w=1` appended to ignore white space changes.

### CODE REVIEW

**Review by weyfonk — COMMENTED** [review]
LGTM apart from a doubt :)

File: internal/cmd/cli/cleanup.go [hunk 1]
```diff
@@ -21,11 +22,20 @@ func NewCleanUp() *cobra.Command {
 }
 
 type CleanUp struct {
+	FleetClient
```
[the remark this answers is not in the document]
Comment (manno) [reply internal/cmd/cli/cleanup.go:25]: This struct is also embedded in the other "wrangler-based" command:
https://github.com/rancher/fleet/pull/2111/files#diff-7cde6ba8e86015b5b406c141d420e4f23ca46f6e1feab5f668dff3b1c1fdb2f6R31

Apply and cleanup still use the wrangler client, more specifically fleet's convenience `client` pkg: https://github.com/rancher/fleet/blob/master/internal/client/client.go

The command line arguments to configure the client are a bit different from the ones used by controller-runtime.

[the remark this answers is not in the document]
Comment (weyfonk) [reply internal/cmd/cli/cleanup.go:25]: Right, so I understand this as a way or moving cobra-set, wrangler-specific client flags closer to the commands that actually use them, as opposed to the root command. Is that correct?

**Review by weyfonk — APPROVED** [review]

File: internal/cmd/cli/cleanup.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (manno) [reply internal/cmd/cli/cleanup.go:25]: Yes, the problem is that the root command flags are always persistent flags. That's how wrangler cli worked. So sub commands inherit flags they don't need. Very confusing with "namespace", "context", etc. when other commands don't use them, or use them differently.

**Pull request merged**